### DISCUSSION TITLE
Where to buy an assembled card?

### DISCUSSION THREAD

**Discussion by vitasam** [markadev/AppleII-VGA#49] [General]
Hi, just found [this](https://www.ebay.com/itm/196506625093?mkcid=16&mkevt=1&mkrid=711-127632-2357-0&ssspo=ti2msstgqo2&sssrc=2047675&ssuid=&widget_ver=artemis&media=COPY) on eBay - is this card an authentic?

**Comment by markadev** [comment]
Hey there, it does look like this is an assembled version of my PCB design and should be compatible with the firmware in this repo.

_Disclaimer: I'm not specifically endorsing this seller. Please use the same caution when buying anything on eBay._

**Reply by vitasam** [reply]
I see. Do you have assembled boards or bare PCB for sale?

**Comment by dhgoss** [comment]
I have some boards in my eBay store, both assembled and plain PCB’s. Also, a new board for HDMI. APPLE IIe VGA Graphics Card Sharp Graphics! Fully Built and Programmedebay.comSent from my iPhoneOn Sep 14, 2024, at 11:35 AM, The Random DIY ***@***.***> wrote:﻿
I see. Do you have assembled boards or bare PCB for sale?

**Reply by vitasam** [reply]
Thanks @dhgoss , can you send a link to your store, I'm interested in the original VGA board?

**Comment by dhgoss** [comment]
Maxwell's Gadgets | eBay Storesebay.comSent from my iPhoneOn Sep 14, 2024, at 12:35 PM, The Random DIY ***@***.***> wrote:﻿
Thanks @dhgoss , can you send a link to your store, I'm interested in the original VGA board?

**Reply by vitasam** [reply]
Thanks! 

Is this card is fully compatible with a firmware from this repository?
https://www.ebay.com/itm/296108805236

**Comment by dhgoss** [comment]
Yes. It’s programmed with firmware from this repository. Sent from my iPhoneOn Sep 14, 2024, at 3:14 PM, The Random DIY ***@***.***> wrote:﻿
Thanks!
Is this card is fully compatible with a firmware from this repository?
https://www.ebay.com/itm/296108805236

**Reply by vitasam** [reply]
Thanks! Do you ship to Finland?

**Comment by dhgoss** [comment]
YesSent from my iPhoneOn Sep 14, 2024, at 3:33 PM, The Random DIY ***@***.***> wrote:﻿
Thanks! Do you ship to Finland?

**Reply by vitasam** [reply]
I got the board, thank you! Hope to test it soon in Apple-IIe

**Comment by vitasam** [comment]
Thanks, ordered

**Reply by vitasam** [reply]
I wonder will it end up here at custom office for the custom clearance

**Comment by vitasam** [comment]
Hello,
Am I right that the card should be installed with a VGA connector toward the keyboard?
![image](https://github.com/user-attachments/assets/8c8b8847-6178-4f50-ad11-b0dfb32bf143)

Does it matter in which slot?

**Reply by dhgoss** [reply]
That is correct

**Reply by dhgoss** [reply]
slot doesn't matter

**Comment by vitasam** [comment]
Hi, the card is alive, but this is shown when I typed "hello": 
![image](https://github.com/user-attachments/assets/fde205a5-d320-466a-918a-53ef67c7a7ce)

Do I need a different firmware for `Apple-IIe` ?

**Comment by dhgoss** [comment]
Yes.   The firmware on your card is for an Apple II.  Go here: https://github.com/markadev/AppleII-VGA/releases/tag/v1.4.0 
Download the .uf2 file for the IIe on your PC.  Hold down the white button on your VGA card and plug it into your PC with a USB cable.   It will show up as an additional drive.  While still holding down the white button drag the .uf2 file to your card.  It will only take a few seconds to load.  Now the card should show the proper characters for a IIe.

**Reply by vitasam** [reply]
Great! Will test it tonight

**Reply by vitasam** [reply]
Hi, the new firmware works well, thanks! 

However I don't fully understand, is this a problem with computer itself, it's ROM or VGA-card. 
E.g. I'm trying Low-res mode colors and pixels BASIC example:

```
05 REM colors and pixels in low-res mode
10 GR : POKE 49234,0 : REM low-res mode, full screen mode 40x48

20 FOR C = 0 TO 15 : REM loop through all 16 colors
30 COLOR=C
40 FOR Y = C * 3 TO (C * 3 + 2)
50 HLIN 0, 39 at Y
60 NEXT 

80 FOR X = 0 TO 15
90 COLOR=X
100 PLOT X * 2 + 4, C*3+1
110 NEXT X

120 NEXT C
130 END
```

The online [emulator](https://www.calormen.com/jsbasic/) shows this:
![image](https://github.com/user-attachments/assets/ddbd9489-0024-483b-93e4-8ce8349be1da)

On my apple:
![image](https://github.com/user-attachments/assets/0656744d-2f0b-4e16-88ae-1dc70842bd67)

**Reply by vitasam** [reply]
last 2.5 color lines are corrupted

**Reply by vitasam** [reply]
Or is this because of that `POKE ... `? UPDATE: no, without POKE the bottom is still corrupted

![image](https://github.com/user-attachments/assets/665b3487-1305-436c-b6b8-1ffade0645fc)

UPDATE 2: the same is without VGA-card, on a composite video

**Reply by markadev** [reply]
Hey there, I'm not sure what could be going on here.  That POKE command is supposed to switch the bottom 4 text lines to graphics and that is how it behaves on my IIe.

I suppose that if the behavior is the same between your composite and VGA output then there's something different in your ROM.  If the behavior between your composite and VGA output is different then that would indicate a problem in the VGA card.

**Reply by vitasam** [reply]
Actually it seems to work, after I've added a `GET A$` delay in the end:
![image](https://github.com/user-attachments/assets/35f10cce-1a13-478f-8e63-593b3ed0512b)

![image](https://github.com/user-attachments/assets/9d5be84a-1cb7-4b81-b12d-856cdd47e608)

So probably it is an example of difference between emulator and the real HW

**Comment by vitasam** [comment]
By the way, will VGA-card work (and support)  if I install that `80-column` memory card in to `Apple-IIe`?

**Reply by markadev** [reply]
Yup it will.  You can run without an 80-column card but it's needed for the double-res video modes that were added to the IIe and most software for the IIe needs the additional 64k of memory that it provides.

**Reply by vitasam** [reply]
Yes, just tested it now with installed 80-column card. Works well.
